### ISSUE TITLE
模型转为diffusers有精度损失

### ISSUE THREAD

**Issue by linwenzhao1** [THUDM/CogVideo#655]
### System Info / 系統信息

diffusers:0.32.dev0
cuda:12.0

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

使用convert_weight_sat2hf.py转化2b全参微调的模型，然后用下面命令加载推理，视频有较大的精度损失
pipe = CogVideoXPipeline.from_pretrained(args.pretrained_model_name_or_path, torch_dtype=torch.bfloat16).to(device)
pipe.scheduler = CogVideoXDPMScheduler.from_config(pipe.scheduler.config, timestep_spacing="trailing")

### Expected behavior / 期待表现

模型转化后无太大精度损失

**Comment by zRzRzRzRzRzRzR** [comment]
这个模型应该用FP16进行转换和推理，在转换的时候，应该用FP32载VAE模块。

**Comment by linwenzhao1** [comment]
> 这个模型应该用FP16进行转换和推理，在转换的时候，应该用FP32载VAE模块。

应该不是数据格式问题，以下是转化代码，还是有精度损失
    dtype=torch.float16
    args.transformer_ckpt_path = "/home/workspace/CogVideo/sat/ckpts_lora_2b/lora-disney_rt1_full_2b_0f-01-04-10-01/9500/mp_rank_00_model_states.pt"
    if args.transformer_ckpt_path is not None:
        init_kwargs = get_transformer_init_kwargs(args.version) ##1.0
        transformer = convert_transformer(
            args.transformer_ckpt_path,
            args.num_layers, ##30
            args.num_attention_heads, ##30
            args.use_rotary_positional_embeddings, ##False
            args.i2v, ##False
            dtype,##fp16
            init_kwargs,
        )
    args.vae_ckpt_path = "/home/CogVideoX-2b-sat/vae/3d-vae.pt"
    if args.vae_ckpt_path is not None:
        # Keep VAE in float32 for better quality
        vae = convert_vae(args.vae_ckpt_path, args.scaling_factor, args.version, torch.float32)

    text_encoder_id = "/home/t5-v1_1-xxl/"
    tokenizer = T5Tokenizer.from_pretrained(text_encoder_id, model_max_length=TOKENIZER_MAX_LENGTH)
    text_encoder = T5EncoderModel.from_pretrained(text_encoder_id, cache_dir=args.text_encoder_cache_dir)

    if args.typecast_text_encoder:
        text_encoder = text_encoder.to(dtype=dtype)

**Comment by linwenzhao1** [comment]
> 跑不起来的报错事？

transformer.load_state_dict(original_state_dict, strict=True)这行代码报算子加载错误，需要先移除模型的conditioner.embedders和vae first_stage_model名称算子，对于cogvideo 5b还需要移除patch_embed.pos_embedding才能加载
